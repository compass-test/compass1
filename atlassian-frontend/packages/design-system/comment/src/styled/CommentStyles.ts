import styled from '@emotion/styled';

import { N100A, N800 } from '@atlaskit/theme/colors';
import { gridSize } from '@atlaskit/theme/constants';
import { token } from '@atlaskit/tokens';

export const Content = styled.div<{ isDisabled?: boolean }>`
  color: ${({ isDisabled }) =>
    isDisabled
      ? token('color.text.disabled', N100A)
      : token('color.text.highEmphasis', N800)};
  margin-top: ${gridSize() / 2}px;
`;
