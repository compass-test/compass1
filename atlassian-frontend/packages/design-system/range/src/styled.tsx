/** @jsx jsx */

/* eslint-disable no-mixed-operators */
import { CSSProperties, forwardRef, useMemo } from 'react';

import { css, jsx } from '@emotion/core';

import { fontFamily } from '@atlaskit/theme/constants';
import { e200 } from '@atlaskit/theme/elevation';

import { ThemeTokens, ThemeTokensTrack } from './theme';

const sliderThumbSize = 16;
const sliderThumbBorderThickness = 2;
const sliderLineThickness = 4;
const transitionDuration = '0.2s';
const sliderBorderRadius = sliderLineThickness / 2;
export const overallHeight = 40;

interface CustomInputProps extends ThemeTokens {
  ref: React.Ref<HTMLInputElement>;
  valuePercent: string;
}

type InputProps = React.InputHTMLAttributes<HTMLInputElement> &
  CustomInputProps;

const getBackgroundGradient = ({ lower, upper }: ThemeTokensTrack) =>
  `
    background: linear-gradient(${lower}, ${lower}) 0 / var(--range-inline-width) 100%
      no-repeat ${upper};
    [dir='rtl'] & {
      background-position: right;
    }
  `;

// Thumb style
const sliderThumbStyle = ({ thumb }: ThemeTokens) => {
  return `
    background: ${thumb.default.background};
    border: ${sliderThumbBorderThickness}px solid transparent;
    border-radius: 50%;
    height: ${sliderThumbSize}px;
    width: ${sliderThumbSize}px;
    box-sizing: border-box;
    transition: border-color ${transitionDuration} ease-in-out;
    ${e200()};
  `;
};

// Track styles
const sliderTrackStyle = `
  border-radius: ${sliderBorderRadius}px;
  border: 0;
  cursor: pointer;
  height: ${sliderLineThickness}px;
  width: 100%;
  transition: background-color ${transitionDuration} ease-in-out;
`;

// Range input styles
const chromeRangeInputStyle = (tokens: ThemeTokens) => {
  return `
  &::-webkit-slider-thumb {
    -webkit-appearance: none;
    margin-top: -${(sliderThumbSize - sliderLineThickness) / 2}px;
    ${sliderThumbStyle(tokens)};
  }

  &:focus::-webkit-slider-thumb {
    border-color: ${tokens.thumb.focus.border};
  }

  &:disabled::-webkit-slider-thumb {
    cursor: not-allowed;
    box-shadow: 0 0 1px ${tokens.thumb.disabled.boxShadow};
  }

  &::-webkit-slider-runnable-track {
    ${sliderTrackStyle};
    ${getBackgroundGradient(tokens.track.default)};
  }

  &:focus::-webkit-slider-runnable-track {
    ${getBackgroundGradient(tokens.track.default)};
  }

  &:active::-webkit-slider-runnable-track,
  &:hover::-webkit-slider-runnable-track {
    ${getBackgroundGradient(tokens.track.hover)};
  }

  &:disabled::-webkit-slider-runnable-track {
    ${getBackgroundGradient(tokens.track.disabled)}
    cursor: not-allowed;
  }
  `;
};

const firefoxRangeInputStyle = (tokens: ThemeTokens) => {
  return `
  &::-moz-focus-outer {
    border: 0;
  }

  &::-moz-range-thumb {
    ${sliderThumbStyle(tokens)};
  }

  &:focus::-moz-range-thumb {
    border-color: ${tokens.thumb.focus.border};
  }

  &:disabled::-moz-range-thumb {
    cursor: not-allowed;
    box-shadow: 0 0 1px ${tokens.thumb.disabled.boxShadow};
  }

  &::-moz-range-progress {
    ${sliderTrackStyle};
    background: ${tokens.track.default.lower};
  }

  &::-moz-range-track {
    ${sliderTrackStyle};
    background: ${tokens.track.default.upper};
  }

  &:active::-moz-range-progress,
  &:hover::-moz-range-progress {
    background: ${tokens.track.hover.lower};
  }

  &:active::-moz-range-track,
  &:hover::-moz-range-track {
    background: ${tokens.track.hover.upper};
  }

  &:disabled::-moz-range-progress {
    background: ${tokens.track.disabled.lower};
    cursor: not-allowed;
  }

  &:disabled::-moz-range-track {
    background: ${tokens.track.disabled.upper};
    cursor: not-allowed;
  }
`;
};

const IERangeInputStyle = (tokens: ThemeTokens) => {
  return `
  &::-ms-thumb {
    margin-top: 0;
    ${sliderThumbStyle(tokens)};
  }

  &:focus::-ms-thumb {
    border-color: ${tokens.thumb.focus.border};
  }

  &:disabled::-ms-thumb {
    cursor: not-allowed;
    box-shadow: 0 0 1px ${tokens.thumb.disabled.boxShadow};
  }

  &::-ms-track {
    background: transparent;
    border-color: transparent;
    color: transparent;
    cursor: pointer;
    height: ${sliderLineThickness}px;
    transition: background-color ${transitionDuration} ease-in-out;
    width: 100%;
  }

  &::-ms-fill-lower {
    background: ${tokens.track.default.lower};
    border-radius: ${sliderBorderRadius}px;
    border: 0;
  }

  &::-ms-fill-upper {
    background: ${tokens.track.default.upper};
    border-radius: ${sliderBorderRadius}px;
    border: 0;
  }

  &:active::-ms-fill-lower,
  &:hover::-ms-fill-lower {
    background: ${tokens.track.hover.lower};
  }

  &:active::-ms-fill-upper,
  &:hover::-ms-fill-upper {
    background: ${tokens.track.hover.upper};
  }

  &:disabled::-ms-fill-lower {
    background: ${tokens.track.disabled.lower};
    cursor: not-allowed;
  }

  &:disabled::-ms-fill-upper {
    background: ${tokens.track.disabled.upper};
    cursor: not-allowed;
  }
`;
};

// Styles are split per browser as they are implemented differently
// Cannot consolidate as Chrome & Firefox don't recognise styles if they are grouped
// with CSS selectors they don't recognise, e.g. &::-ms-thumb
export const rangeInputStyle = (tokens: ThemeTokens) => {
  return css`
    -webkit-appearance: none; /* Hides the slider so that custom slider can be made */
    background: transparent; /* Otherwise white in Chrome */
    height: ${overallHeight}px; /* Otherwise thumb will collide with previous box element */
    padding: 0; /* IE11 includes padding, this normalises it */
    width: 100%; /* Specific width is required for Firefox. */

    &:focus {
      outline: none;
    }

    &:disabled {
      cursor: not-allowed;
    }

    ${chromeRangeInputStyle(tokens)}
    ${firefoxRangeInputStyle(tokens)}
    ${IERangeInputStyle(tokens)};

    font-family: ${fontFamily()};

    background-position: right;
  `;
};

export const Input = forwardRef(
  (props: InputProps, ref: React.Ref<HTMLInputElement>) => {
    const { valuePercent, thumb, track, style, ...strippedProps } = props;

    // Note: emotion will cache unique outputs as their own this
    // We are memoizing the creation of this string
    const styles = useMemo(() => rangeInputStyle({ track, thumb }), [
      thumb,
      track,
    ]);

    // We are creating a css variable to control the "progress" portion of the range input
    // This avoids us needing to create a new css class for each new percentage value
    return (
      <input
        {...strippedProps}
        style={{ '--range-inline-width': `${valuePercent}%` } as CSSProperties}
        ref={ref}
        css={styles}
      />
    );
  },
);

Input.displayName = 'InputRange';
