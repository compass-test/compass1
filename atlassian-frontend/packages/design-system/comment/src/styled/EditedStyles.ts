import styled from '@emotion/styled';

import { N200 } from '@atlaskit/theme/colors';
import { token } from '@atlaskit/tokens';

export default styled.span`
  color: ${token('color.text.lowEmphasis', N200)};
`;
