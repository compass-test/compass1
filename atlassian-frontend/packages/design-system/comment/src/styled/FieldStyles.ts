import { css } from '@emotion/core';
import styled from '@emotion/styled';

import { N500 } from '@atlaskit/theme/colors';
import { token } from '@atlaskit/tokens';

interface CommonProps {
  hasAuthor?: boolean;
}
const common = ({ hasAuthor }: CommonProps) => css`
  &:not(:hover):not(:active) {
    color: ${token('color.text.mediumEmphasis', N500)};
  }
  font-weight: ${hasAuthor ? 500 : 'inherit'};
`;

export const Anchor = styled.a<CommonProps>`
  ${(p) => common(p)};
`;
export const Span = styled.span<CommonProps>`
  ${(p) => common(p)};
`;
