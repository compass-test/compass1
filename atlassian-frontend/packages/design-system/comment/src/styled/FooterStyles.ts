import styled from '@emotion/styled';

import { N500, Y500 } from '@atlaskit/theme/colors';
import { gridSize } from '@atlaskit/theme/constants';
import { token } from '@atlaskit/tokens';

import { actionsPadding } from './constants';

export const ActionsItem = styled.div`
  display: flex;

  & + &::before {
    color: ${token('color.text.mediumEmphasis', N500)};
    content: 'Â·';
    display: inline-block;
    text-align: center;
    vertical-align: middle;
    width: ${actionsPadding}px;
  }
`;

export const ErrorIcon = styled.span`
  color: ${token('color.text.warning', Y500)};
  padding-right: ${gridSize()}px;
`;

export const ActionsContainer = styled.div`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  margin-top: ${gridSize() * 0.75}px;
`;
