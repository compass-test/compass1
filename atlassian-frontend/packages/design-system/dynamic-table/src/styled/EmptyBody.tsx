import styled from 'styled-components';

import { gridSize } from '@atlaskit/theme/constants';

export const EmptyViewWithFixedHeight = styled.div`
  height: ${gridSize() * 18}px;
`;

export const EmptyViewContainer = styled.div`
  margin: auto;
  padding: 10px;
  text-align: center;
  width: 50%;
`;
