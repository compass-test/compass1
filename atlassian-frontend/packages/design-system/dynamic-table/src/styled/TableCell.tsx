import styled from 'styled-components';

import { cellStyle, truncateStyle, TruncateStyleProps } from './constants';

export const TableBodyCell = styled.td<TruncateStyleProps>`
  ${(props) => truncateStyle(props)} ${cellStyle};
`;
