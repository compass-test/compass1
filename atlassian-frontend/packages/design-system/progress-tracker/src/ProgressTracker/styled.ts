import styled from 'styled-components';

export const ProgressTrackerContainer = styled.ul`
  display: flex;
  width: 100%;

  && {
    margin-top: 40px;
  }

  list-style-type: none;
  padding: 0px;
`;

export default ProgressTrackerContainer;
