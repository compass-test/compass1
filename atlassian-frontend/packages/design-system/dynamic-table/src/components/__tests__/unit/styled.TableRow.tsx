import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';

import { TableBodyRow } from '../../../styled/TableRow';

describe('TableRow', () => {
  let shallowWrapper: ShallowWrapper;

  describe('on rendering', () => {
    beforeEach(() => {
      shallowWrapper = shallow(<TableBodyRow />);
    });

    it('should render default background color', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });

    it('should render no outline without focus', () => {
      expect(shallowWrapper).not.toHaveStyleRule('outline');
      expect(shallowWrapper).not.toHaveStyleRule('outline-offset');
    });

    it('should render blue outline when focused on', () => {
      const focusOption = { modifier: ':focus' };

      expect(shallowWrapper).toHaveStyleRule(
        'outline',
        '2px solid #4C9AFF',
        focusOption,
      );
      expect(shallowWrapper).toHaveStyleRule(
        'outline-offset',
        '-2px',
        focusOption,
      );
    });
  });

  describe('on rendering with highlighted prop', () => {
    beforeEach(() => {
      shallowWrapper = shallow(<TableBodyRow isHighlighted />);
    });

    it('should render with highlighted background color', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
