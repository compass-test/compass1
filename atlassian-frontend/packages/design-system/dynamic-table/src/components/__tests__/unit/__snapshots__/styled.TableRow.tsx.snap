// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow on rendering should render default background color 1`] = `
.c0:hover {
  background-color: #FAFBFC;
}

.c0:focus {
  outline: 2px solid #4C9AFF;
  outline-offset: -2px;
}

<tr
  className="c0"
/>
`;

exports[`TableRow on rendering with highlighted prop should render with highlighted background color 1`] = `
.c0 {
  background-color: #DEEBFF;
}

.c0:hover {
  background-color: #B3D4FF;
}

.c0:focus {
  outline: 2px solid #4C9AFF;
  outline-offset: -2px;
}

<tr
  className="c0"
/>
`;
